system/src/system_network_internal.h

```diff
@@ -416,13 +483,19 @@ class ManagedNetworkInterface : public NetworkInterface
                 WLAN_CONNECTING = 1;
                 INFO("ARM_WLAN_WD 1");
                 ARM_WLAN_WD(CONNECT_TO_ADDRESS_MAX);    // reset the network if it doesn't connect within the timeout
+                const auto diag = NetworkDiagnostics::instance();
+                diag->status(NetworkDiagnostics::CONNECTING);
                 system_notify_event(network_status, network_status_connecting);
```

Comment:
I would prefer to keep the diagnostics stuff separated from the system event logic for now, since the current eventing system is not designed for anything else than notifying the application code of system events (all handlers are processed in the application thread, user handlers may introduce an arbitrary delay, etc). Definitely it would be nice to track all the states defined by the system in a centralized manner, it just feels more like a task that should be done as part of the system loop refactoring

Reply:
👍